uint32 setting
uint32 value
---
int32 return_code